platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultFileLockManager.java

```diff
@@ -286,6 +286,25 @@ public LockMode getMode() {
             return mode;
         }
 
+        /**
+         * Lock file is {@link java.io.RandomAccessFile} that has two regions:
+         * - lock state region, locked for the duration of the operation
+         * - lock info region, locked just to write the lock info or read info from it<br><br>
+         *
+         * We first try to acquire a lock on the state region with retries, see {@link #lockStateRegion(LockMode)}.
+         * If we use exclusive lock, and we succeed, we acquire a exclusive lock on the information region and write our details (port and lock id)
+         * there, and then we release lock of information region. That way other processes can read our details and ping us.
+         * That is important for {@link org.gradle.cache.FileLockManager.LockMode.OnDemand} mode.
+         * If we use shared lock, and we succeed we don't write anything to information region, and thus we currently don't have on demand shared locks.
+         * If we fail, we throw a timeout exception.<br><br>
+         *
+         * What happens in {@link #lockStateRegion(LockMode)}:<br>
```

Comment:
🤔 Why not add this to `lockStateRegion()`?

Reply:
Added to `lockStateRegion()`